scripts/serie_update.py

```diff
@@ -0,0 +1,674 @@
+import logging
+
+STM32_CUBE_REPO_BASE = "https://github.com/STMicroelectronics/STM32Cube"
+
+
+#################################
+# Remove read only protection
+#
+def remove_readonly(func, path, _):
+    os.chmod(path, stat.S_IWRITE)
+    func(path)
+
+
+#################################
+# Execute a command from shell
+#
+def os_cmd(cmd):
+    logging.info(cmd)
+    return subprocess.call(cmd, shell=True)
+
+
+#################################
+# class Stm32SerieUpdate
+#
+class Stm32SerieUpdate:
+    def __init__(self, stm32_serie, stm32cube_repo_path, force, noclean):
+        if not stm32_serie.startswith("stm32"):
+            raise Exception(
+                "Error: Unknown stm32 serie: "
+                + stm32_serie
+                + ". Must start with 'stm32'"
+            )
+
+        # Set serie variables
+        self.stm32_serie = stm32_serie
+        self.stm32_seriexx = stm32_serie + "xx"  # ex:stm32f3xx
+        self.stm32_serie_upper = stm32_serie.upper()  # ex:STM32F3
+        self.stm32_seriexx_upper = self.stm32_serie_upper + "xx"  # ex:STM32F3xx
+        self.serie = self.stm32_serie_upper[5:]
+        self.force = force
+        self.noclean = noclean
+
+        # #####  3 root directories to work with ########
+        # 1: STM32Cube repo Default $HOME/STM32Cube_repo
+        # 2 : zephyr stm32 path  : ex: .../zephyr_project/module/hal/stm32
+        # 3: Temporary directory to construct the update
+        # (within STM32Cube repo dir)
+        self.stm32cube_repo_path = stm32cube_repo_path
+        if not self.stm32cube_repo_path.exists():
+            self.stm32cube_repo_path.mkdir()
+
+        self.zephyr_hal_stm32_path = (
+            Path(os.getenv("ZEPHYR_BASE")).absolute()
+            / ".."
+            / "modules"
+            / "hal"
+            / "stm32"
+        )
+        if not self.zephyr_hal_stm32_path.exists():
+            raise Exception("Error: cannot find zephyr project")
+
+        self.stm32cube_temp = self.stm32cube_repo_path / "temp_stm32xx_update"
+        if self.stm32cube_temp.exists():
+            shutil.rmtree(str(self.stm32cube_temp), onerror=remove_readonly)
+        self.stm32cube_temp.mkdir()
+
+        # subdir specific to a stm32 serie
+        self.stm32cube_serie_path = self.stm32cube_repo_path / Path(
+            "STM32Cube" + self.serie
+        )
+        self.zephyr_module_serie_path = (
+            self.zephyr_hal_stm32_path / "stm32cube" / self.stm32_seriexx
+        )
+        self.stm32cube_temp_serie = (
+            self.stm32cube_temp / "stm32cube" / self.stm32_seriexx
+        )
+        shutil.rmtree(str(self.stm32cube_temp), onerror=remove_readonly)
+        self.stm32cube_temp_serie.mkdir(parents=True)
+
+        self.readme_file_path = self.zephyr_module_serie_path / "README"
+        self.version_tag = []
+        self.current_version = ""
+        self.latest_version = ""
+        self.latest_commit = ""
+
+        logging.basicConfig(level=logging.INFO)
```

Comment:
logger should be configured out of the class.

Reply:
Done